packages/webapp/webapp_server.js

```diff
@@ -345,8 +345,33 @@ function getBoilerplate(request, arch) {
   return getBoilerplateAsync(request, arch).await();
 }
 
+WebApp.encodeRuntimeConfig = function (rtimeConfig) {
+  return JSON.stringify(encodeURIComponent(JSON.stringify(rtimeConfig)));
+}
+
+WebApp.decodeRuntimeConfig = function (rtimeConfig) {
+  return JSON.parse(decodeURIComponent(JSON.parse(rtimeConfig)));
+}
+
+const runtimeConfig = {
+  'hooks': [],
+  'update': {}
+};
+
+WebApp.addRuntimeConfigHook = function (hook) {
+  if(typeof hook !== 'function') throw new Error('WebApp.addRuntimeConfigHook must be a function');
+  runtimeConfig.hooks.push(hook);
+}
+
 function getBoilerplateAsync(request, arch) {
-  const boilerplate = boilerplateByArch[arch];
+  let boilerplate = boilerplateByArch[arch];
+  _.each(runtimeConfig.hooks, function (hook) {
```

Comment:
We are avoiding `underscore`. This would be better
```suggestion
runtimeConfig.hooks.forEach(function (hook) {
```

Reply:
Thanks! Done.